lib/portlayer/storage/vsphere/image.go

```diff
@@ -271,6 +272,60 @@ func (v *ImageStore) WriteImage(op trace.Operation, parent *portlayer.Image, ID
 	return newImage, nil
 }
 
+// ReadArchive reads the delta between child and parent image layers, returning
+// the difference as a tar archive.
+//
+// store - the image store containing the two layers
+// child - the leaf layer of the image
+// ancestor - the layer up the chain against which to diff
+// pathspec - describes filters on paths found in the data (include, exclude, strip)
+// data - set to true to include file data in the tar archive, false to include headers only
+func (v *ImageStore) ReadArchive(op trace.Operation, store *url.URL, child, ancestor string, pathspec map[string]string, data bool) (io.ReadCloser, error) {
+	storeName, err := util.ImageStoreName(store)
+	if err != nil {
+		return nil, err
+	}
+
+	if ancestor == "" {
+		ancestor = portlayer.Scratch.ID
+	}
+
+	c := v.imageDiskDSPath(storeName, child)
+	a := v.imageDiskDSPath(storeName, ancestor)
+
+	cleanFunc := func() {
+		if err := v.dm.UnmountAndDetach(op, c); err != nil {
+			op.Infof("Error cleaning up child disk: %s", err.Error())
+		}
+		if err = v.dm.UnmountAndDetach(op, a); err != nil {
+			op.Infof("Error cleaning up ancestor disk: %s", err.Error())
+		}
+	}
+
+	childFs, err := v.dm.AttachAndMount(op, c)
+	if err != nil {
+		return nil, err
+	}
+
+	ancestorFs, err := v.dm.AttachAndMount(op, a)
+	if err != nil {
+		cleanFunc()
+		return nil, err
+	}
+
+	tar, err := archive.Diff(op, childFs, ancestorFs, pathspec, data)
```

Comment:
May need a filesystem object to get involved contributing to the pathspec - for example "lost+found" should probably not show up in the changeset.